module/VuFind/src/VuFind/Search/Factory/AbstractSolrBackendFactory.php

```diff
@@ -542,4 +552,17 @@ protected function getInjectConditionalFilterListener(Config $search)
         );
         return $listener;
     }
+
+    /**
+     * Get a default parameters listener for the backend
+     *
+     * @param Backend $backend Search backend
+     * @param array   $params  Default parameters
+     *
+     * @return DeduplicationListener
+     */
+    protected function getDefaultParametersListener(Backend $backend, array $params)
```

Comment:
This reminds me that I'd really like to come up with a more elegant way of managing listeners, since this factory just keeps getting bigger and bigger... but that's a problem for another day. :-)

Reply:
Indeed. A plugin manager might be nice...

Reply:
I think the obstacle to a plugin manager is that many of the listeners are constructed with backend-specific constructor parameters that are specific to the factory. We need a distinct instance of each listener for each backend. That would either require a separate plugin manager per backend, a non-standard naming convention for retrieving listeners from the plugin manager, or a refactoring of the listeners to have standardized constructor parameters and accept configuration after construction. None of these solutions seem ideal, though some might be better than what we currently have!

Reply:
You're of course right on all accounts.